src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,35 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../services/user';
+import { User } from '../../types/User';
+
+type Props = {
+  todo: Todo;
+  setTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
+      .then(data => {
+        setUser(data);
+      })
+      .catch(() => {
+        setUser(null);
+      })
+      .finally(() => setTimeout(() => setIsLoading(false), 300));
```

Comment:
The use of `setTimeout` to delay setting `isLoading` to `false` might not be necessary. Consider removing it unless there's a specific reason for the delay.